lib/download.rb

```diff
@@ -0,0 +1,75 @@
+# Code that translates scopes into downloadable tab-delimited CSV.
+module Download
+
+  # @params exclude_columns [Array] delete referenced columns from result 
+  # @params trim_rows [Array] delete rows from result if no data provided
+  # @params trim_columns [Array] delete columns from result if no data provided
+   # @return [String]
+  #   translate a scope into a CSV table, with optional tweaks to the data
+  def self.generate_csv(scope, exclude_columns: [], trim_rows: false, trim_columns: false)
+    column_names = scope.columns_hash.keys
+    headers = CSV::Row.new(column_names, column_names, true)
+    tbl = CSV::Table.new([headers])
+    
+    scope.find_each do |o|
+      tbl << o.attributes.values_at(*column_names).collect{|v| Utilities::Strings.sanitize_for_csv(v) }
```

Comment:
See the comment for Utilities::String.sanitize_for_csv, but here you could explore not sanitize at all and let Ruby's csv library do that for you (but again see the other comment).